pkg/virt-handler/virtwrap/manager.go

```diff
@@ -267,7 +268,9 @@ func NewConnection(uri string, user string, pass string, checkInterval time.Dura
 		callbacks:     make([]libvirt.DomainEventLifecycleCallback, 0),
 		reconnectLock: &sync.Mutex{},
 	}
-	lvConn.installWatchdog(checkInterval)
+	if checkInterval > 0 {
+		lvConn.installWatchdog(checkInterval)
```

Comment:
That does not make sense, we need the watchdog to ensure proper reconnect behaviour. Think about the case where you are not create or modify VMs for 30 minutes. In this case, virt-handler could at the worst case not detect for 30 minutes that it is not connected to libvirt, and we would miss domain event updates for that period. I would check if it is less than maybe 1 second and return an error ...